workers/unity/Packages/io.improbable.gdk.testutils/Editor/SpatialdManager.cs

```diff
@@ -0,0 +1,199 @@
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Threading.Tasks;
+using Improbable.Gdk.Tools;
+using Improbable.Gdk.Tools.MiniJSON;
+using UnityEngine;
+
+namespace Improbable.Gdk.TestUtils.Editor
+{
+    public class SpatialdManager : IDisposable
+    {
+        private static readonly string SpotBinary;
+        private static readonly string ProjectName;
+
+        static SpatialdManager()
+        {
+            var spotPath = Path.Combine(Common.GetPackagePath("io.improbable.worker.sdk"), ".spot/spot");
+
+            if (Application.platform == RuntimePlatform.WindowsEditor)
+            {
+                spotPath = Path.ChangeExtension(spotPath, ".exe");
+            }
+
+            SpotBinary = spotPath;
+
+            var spatialJsonFile = Path.Combine(Tools.Common.SpatialProjectRootDir, "spatialos.json");
+
+            if (!File.Exists(spatialJsonFile))
+            {
+                throw new FileNotFoundException($"Could not find a spatialos.json file located at: {spatialJsonFile}");
+            }
+
+            var data = Json.Deserialize(File.ReadAllText(spatialJsonFile));
+
+            if (data == null)
+            {
+                throw new FormatException($"Could not parse spatialos.json file located at: {spatialJsonFile}");
```

Comment:
Not 100% on using `FormatException` for this...

Reply:
I sort of took a loose interpretation of: 

> In a call to a method that converts a string to some other data type

to mean JSON string -> `spatialos.json` format.

Is there a standard library exception that fits better do you feel? Or should I just create a custom one